### PR TITLE
updated mail tests to be run in parallel

### PR DESCRIPTION
**Pull request by palmieric**
This PR is meant to remove the need to delete all the emails in mailhog before and after a test run so now tests can be performed in parallel.

### CODE REVIEW

File: .github/workflows/acceptance.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 name: acceptance
 on:
   push:
-    branches: [ main ]
```
Comment (mganisin) [inline .github/workflows/acceptance.yml:4]: 8 commits in total, plus commits that shouldn't be here at all, this requires redesign

**Review by mganisin — CHANGES_REQUESTED** [review]

File: .github/workflows/acceptance.yml (hunk 1 above)
Comment (palmieric) [reply .github/workflows/acceptance.yml]: done

File: testsuite/tests/system/messages/mail_templates.yml [hunk 2]
```diff
@@ -25,15 +25,15 @@ subject_templates:
       Body: '----==_mimepart_.+Content-Type: text/plain; charset=UTF-8Content-Transfer-Encoding:
       7bitDear <username>,A new application subscribed to the <aplan> plan on the
       <service> service of the <test_group> account.Application details:  Name:
-      API signup  Description: API signupTo view application, follow this link: https://3scale-admin.<threescale_superdomain>/apiconfig/services/[0-9]+/applications/[0-9]+You
+      API signup  Description: API signupTo view application, follow this link: https://3scale-admin.<threescale_superdomain>/p/admin/applications/[0-9]+You
```
Comment (palmieric) [inline testsuite/tests/system/messages/mail_templates.yml:28]: @mganisin these changes will break the tests for the previous versions of 3scale, how could we proceed here?

Comment (pehala) [reply testsuite/tests/system/messages/mail_templates.yml:28]: You could theoretically create a new template file for 2.11 and choose which one to use depending on the tested version. Not sure if it is the preferred solution tho.

Comment (mganisin) [reply testsuite/tests/system/messages/mail_templates.yml:28]: discussed live and found an agreement

**Comment by pehala** [comment]
```
mailhog_client = <testsuite.mailhog.MailhogClient object at 0x7f266829fdf0>
mail_template = {'equal_templates': {'From': '"a-213.apps.ocp-cluster.osp.api-qe.eng.rdu2.redhat.com Notification" <no-reply@a-213.app...id":[0-9]+,"account_id":[0-9]+},"category":\\["notification","service_contracts/service_contract_created_event"\\]}'}}}

    def matching_emails(mailhog_client, mail_template):
        """
        Checks that the 'To', 'From' and 'Return-Path' addresses match the addresses from
        the template
        Checks that the message body and header matches one of the items in the template that
        was not already matched. (The sent emails shouldn't be identical)
        Returns the ids of the matching emails
        """
        ids = []
        messages = mailhog_client.messages()
        checked_messages = []
        for message in messages['items']:
            message_body = message["Content"]["Body"]\
                .replace("=\r\n", "").replace("\r\n", "")
            headers = message["Content"]["Headers"]
    
            for address_type in ["To", "From", "Return-Path"]:
>               if headers[address_type][0] != mail_template["equal_templates"][address_type]:
E               KeyError: 'To'

testsuite/tests/system/messages/test_email_accounts.py:52: KeyError
```
I keep getting this error when running the tests in parallel @palmieric

**Comment by palmieric** [comment]
> ```
> mailhog_client = <testsuite.mailhog.MailhogClient object at 0x7f266829fdf0>
> mail_template = {'equal_templates': {'From': '"a-213.apps.ocp-cluster.osp.api-qe.eng.rdu2.redhat.com Notification" <no-reply@a-213.app...id":[0-9]+,"account_id":[0-9]+},"category":\\["notification","service_contracts/service_contract_created_event"\\]}'}}}
> 
>     def matching_emails(mailhog_client, mail_template):
>         """
>         Checks that the 'To', 'From' and 'Return-Path' addresses match the addresses from
>         the template
>         Checks that the message body and header matches one of the items in the template that
>         was not already matched. (The sent emails shouldn't be identical)
>         Returns the ids of the matching emails
>         """
>         ids = []
>         messages = mailhog_client.messages()
>         checked_messages = []
>         for message in messages['items']:
>             message_body = message["Content"]["Body"]\
>                 .replace("=\r\n", "").replace("\r\n", "")
>             headers = message["Content"]["Headers"]
>     
>             for address_type in ["To", "From", "Return-Path"]:
> >               if headers[address_type][0] != mail_template["equal_templates"][address_type]:
> E               KeyError: 'To'
> 
> testsuite/tests/system/messages/test_email_accounts.py:52: KeyError
> ```
> 
> I keep getting this error when running the tests in parallel @palmieric

Do you know if the key error is from headers or mail_template["equal_templates"]?
I never got this kind of error to be honest.

**Comment by pehala** [comment]
LGTM!

**Review by pehala — APPROVED** [review]

**Review by mganisin — APPROVED** [review]

**Pull request merged**